torch/_higher_order_ops/while_loop.py

```diff
@@ -246,16 +328,56 @@ def _trace_while_loop(
 
 
 def check_outputs_carry_consistency(
-    outs: List[torch.Tensor], carries: List[torch.Tensor]
+    outs: List[Union[torch.Tensor, torch.SymInt, int]],
+    carries: List[Union[torch.Tensor, torch.SymInt, int]],
 ) -> None:
-    all_diffs_in_meta = []
-    for out, cry in zip(outs, carries):
-        if diff := diff_tensor_meta(
-            _extract_tensor_metadata(cry), _extract_tensor_metadata(out)
-        ):
-            all_diffs_in_meta.append(",".join(diff))
-    if all_diffs_in_meta:
-        diff_str = "\n".join(all_diffs_in_meta)
+    def diff_meta_pairs(
+        lhs_list: List[Union[torch.Tensor, torch.SymInt, int]],
+        rhs_list: List[Union[torch.Tensor, torch.SymInt, int]],
+    ) -> List[str]:
+        def diff_meta(
+            lhs: Union[torch.Tensor, torch.SymInt, int],
+            rhs: Union[torch.Tensor, torch.SymInt, int],
+        ) -> str:
+            if isinstance(lhs, torch.Tensor) and isinstance(rhs, torch.Tensor):
+                return ", ".join(
+                    diff_tensor_meta(
+                        # We set include contiguity=False because we have vmap x cond tests, where if
+                        # include_contiguity=True will call t.is_contiguous inside of vmap and get an error
+                        # "querying is_contiguous inside of vmap for memory_format other than
+                        # torch.contiguous_format is not yet implemented". This is good for now.
+                        _extract_tensor_metadata(lhs, include_contiguity=False),
+                        _extract_tensor_metadata(rhs, include_contiguity=False),
```

Comment:
I don't think this is OK? We want to make sure the outputs have the same contiguity. Alternatively, you can force the outputs to all be contiguous

Reply:
This happens in case like this:
```python
    def test_cond_vmap_multiple_inputs(self):
        def fn(x, y):
            return torch.cond(
                pred=x.sum() < y.sum(),
                true_fn=lambda x, y: x + 100,
                false_fn=lambda x, y: y,
                operands=(x, y),
            )

        a = torch.arange(15).reshape(3, 5)
        b = torch.ones_like(a) + 3
        res = torch.vmap(fn, in_dims=(0, 0))(a, b)
```

Where we check the memory formats  for true_r and false_r after speculate subgraph of cond, where the outputs are BatchedTensors:
```python
        memory_formats = {
            torch.contiguous_format,
            torch.channels_last,
            torch.channels_last_3d,
        }
        for query_format in memory_formats:
            if result.is_contiguous(memory_format=query_format):
                memory_format = query_format
                break
```
After set include_contiguity=False, there are three format types to be checked against torch.is_contiguous() but we only support querying `torch.contiguous_format` for BatchedTensor. Otherwise, we get: "NYI: querying is_contiguous inside of vmap for memory_format other than torch.contiguous_format". 

Is it simple to support querying is_contiguous for batched tensor for other formats? Some workarounds might be we only check the contiguous format if it's batched tensor, or we check all three formats for the FakeTensor wrapped in the batched tensor (can we do that?is the FakeTensor inside BatchedTensor a slice of original tensor?).